dev/benchmarks/macrobenchmarks/lib/src/large_images.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+class LargeImagesPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final ImageCache imageCache = PaintingBinding.instance.imageCache;
+    imageCache.maximumSize = 30;
+    imageCache.maximumSizeBytes = 50 << 20;
+    return GridView.builder(
+      itemCount: 1000,
+      gridDelegate:
+      const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 3),
+      itemBuilder: (BuildContext context, int index) {
+        return Image.network(
```

Comment:
Can we avoid using a network image for this?  If the server goes down our test will fail right?

Reply:
I tried using local images in https://github.com/flutter/flutter/compare/master...liyuqian:local_image_test but unfortunately it no longer reproduces the memory issue. It seems that some image downloading or decompressing is involved in this memory issue.

I just strengthened the test to scroll even faster so we can see a 500MB+ (sometimes 700MB) memory usage. One more scroll crashed the test in my Moto G4.

Reply:
What if we do it with a asset image that isn't cached so we have to hit the disk each time we access it?

Or what if we made an image provider that simulates a delay as if it was coming from the network each time?

If we _really_ need to hit the network for this, we should probably see if we can hit a domain that we're more confident in or need to have up anyway (like some image from pub.dev or flutter.dev).